subsys/bluetooth/controller/ll_sw/lll_filter.h

```diff
@@ -4,27 +4,31 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_BT_CTLR_FAL_SIZE)
-#define FAL_SIZE CONFIG_BT_CTLR_FAL_SIZE
+#if defined(CONFIG_BT_CTLR_FAL_SIZE) && defined(CONFIG_BT_CTLR_RL_SIZE)
+#define LLL_FILTER_SIZE MAX(CONFIG_BT_CTLR_FAL_SIZE, CONFIG_BT_CTLR_RL_SIZE)
+#elif defined(CONFIG_BT_CTLR_FAL_SIZE)
+#define LLL_FILTER_SIZE CONFIG_BT_CTLR_FAL_SIZE
+#elif defined(CONFIG_BT_CTLR_RL_SIZE)
+#define LLL_FILTER_SIZE CONFIG_BT_CTLR_RL_SIZE
 #else
-#define FAL_SIZE 8
-#endif /* CONFIG_BT_CTLR_FAL_SIZE */
+#define LLL_FILTER_SIZE 8
+#endif /* CONFIG_BT_CTLR_FAL_SIZE && CONFIG_BT_CTLR_RL_SIZE */
 
 #define IRK_SIZE               16
 #define FILTER_IDX_NONE        0xFF
-#define LLL_FILTER_BITMASK_ALL (BIT(FAL_SIZE) - 1)
+#define LLL_FILTER_BITMASK_ALL (BIT(LLL_FILTER_SIZE) - 1)
 
 struct lll_filter {
-#if (FAL_SIZE <= 8)
+#if (LLL_FILTER_SIZE <= 8)
```

Comment:
```suggestion
#if (LLL_FILTER_SIZE <= CHAR_BIT)
```

Reply:
Should we do it if type is `uint8_t` (which explicitly denotes that it has 8 bits)?
CHAR_BIT is not guaranteed to have 8 bits on some architectures

Reply:
`CHAR_BIT >= 8`, if it is greater than 8, then uint8_t should not exist for that particular architecture (at least to my humble opinion).
You can go with this: `#define UCHAR_WIDTH (sizeof(uint8_t) * 8)` that is provided in C23 standard.

Reply:
Discussed online. Applied changes.